Assets/VRTK/Scripts/VRTK_ThirdPersonCamera.cs

```diff
@@ -0,0 +1,113 @@
+﻿// Third Person Camera|Scripts|0090
+namespace VRTK
+{
+    using UnityEngine;
+    /// <summary>
+    /// This script will follow and switch between camera targets that are attached to a third person character. Attach to `[CameraRig]` and then set the array of game objects that should be attached to the empty root object of the 3rd person character
+    /// </summary>
+    /// <remarks>
+    /// This script is intended to be paired with GazeDrivenLocomotion
+    /// </remarks>
+    public class VRTK_ThirdPersonCamera : MonoBehaviour
+    {
+        [Tooltip("The speed the camera should move at it's max speed")]
+        [SerializeField]
+        private float cameraMoveSpeed = 1.0f;
+        [Tooltip("An array of transforms that represent different camera angles. Must have at least 1 transform for this script to work. It is important to attach the transform to the empty root object that is above the character mesh.")]
+        [SerializeField]
+        private Transform[] targetArray;
+        [Tooltip("Because acceleration can sometimes cause simulator sickness, we wanted to try changing the acceleration curve. One easy way to do that is with animation curves. The animation curve can make the acceleration curve slower to curb simulator sickness by having the 'Y' axis grow slower.")]
+        [SerializeField]
+        private AnimationCurve animationCurve;
+        [Tooltip("Determines whether to use animation curve or standard lerp mechanism. Don't enable unless you've created an animation curve")]
+        [SerializeField]
+        private bool useAnimationCurve;
+        [Tooltip("Used to switch camera angles when the grip button is pressed. Remember to assign VRTK_ControllerEvents as a component of your left controller.")]
+        [SerializeField]
+        private VRTK_ControllerEvents leftControllerEvents;
+        [Tooltip("Used to switch camera angles when the grip button is pressed. Remember to assign VRTK_ControllerEvents as a component of your right controller.")]
+        [SerializeField]
+        private VRTK_ControllerEvents rightControllerEvents;
+        private Vector3 targetPreviousPosition;
+        private Transform target;
+        private float animationCurvePercentage = 0;
+        private GameObject currentTransformObject;
+        private int currentIndex = 0;
+        private Transform hmd;
+        // Finds HMD position, then sets the transform of the camera
+        void Start()
```

Comment:
All MonoBehaviour methods need to be `protected virtual` so people can extend the scripts if required